scrunch/cubes.py

```diff
@@ -1,17 +1,33 @@
+import six
+
 from pycrunch.cubes import fetch_cube, count
 from scrunch.datasets import Variable
+from scrunch.expressions import parse_expr, process_expr
 from cr.cube.crunch_cube import CrunchCube
 
 
-def crtabs(dataset, variables):
+def crtabs(dataset, variables, weight=None, filter=None, **measures):
     """Return CrunchCube representation of crosstab.
 
     :param dataset: Dataset shoji object
     :param variables: List of variable urls, names or aliases
+    :param weight: Scrunch variable instance, alias or url
+    :param filter: Scrunch expression filter
     """
-    dataset = dataset.resource
     variables = prepare_variables(variables)
-    return CrunchCube(fetch_cube(dataset, variables, count=count()))
+    if weight:
+        if isinstance(weight, Variable):
+            weight = weight.url
+        elif isinstance(weight, six.string_types):
+            if weight in dataset.keys():
+                weight = dataset[weight].url
+        else:
+            raise KeyError('{} not found in dataset'.format(weight))
```

Comment:
@anderstheet: a better error would be to throw `weight` argument type not supported or something along those lines. If a weight is not a Variable or a string it shouldn't be a KeyError, perhaps ValueError.

Reply:
It will work on both cases, if the variable alias is not in the dataset keys then we can assume it's a url and pass that to the API, it's practice in scrunch to support variable arguments as alias, instance or url, otherwise the API should return the error.